src/Hyperion/Benchmark.hs

```diff
@@ -55,7 +60,37 @@ instance Show Benchmark where
   showsPrec x (WithSampling _opt bk) = showsPrec x bk
 
 bench :: String -> Batch () -> Benchmark
-bench name batch = Bench (Text.pack name) batch
+bench name batch = Bench Text.pack (const []) id (const batch) name
+
+benchWith
+  :: (a -> Text)
+  -> (a -> [JSON.Pair])
```

Comment:
And even if we decide we do want arbitrary user metadata, I'd rather a type closer to `benchMany2` than the type of `benchWith`. At least `benchMany2` (nearly!) doesn't hard code JSON (it does hardcode, but it's easy to get rid of the `ToJSON` instance and move it elsewhere).